theano/tensor/opt.py

```diff
@@ -3885,8 +3890,8 @@ def local_sum_mul_by_scalar(node):
                         return [T.mul(scalars[0], node.op(non_scalars[0]))]
                     else:
                         return [scalars[0]]
-        if thing_summed.owner and thing_summed.owner.op == T.neg:
-            return [T.neg(node.op(thing_summed.owner.inputs[0]))]
+        if node_inps.owner and node_inps.owner.op == T.neg:
+            return [T.neg(node.op(node_inps.owner.inputs[0]))]
```

Comment:
Are you sure that -prod(smth) == prod(-smth).  I think the sign changes according to the number of elements, so we should skip this replacement for Prod.